model.lua

```diff
@@ -150,33 +150,57 @@ function Model:retrieve(dataloader, dtype)
 
     self.params.numOptions = 100;
 
-    -- ranks for the given datatype
-    local ranks = torch.Tensor(numThreads, self.params.maxQuesCount);
-    ranks:fill(self.params.numOptions + 1);
-
-    while startId <= numThreads do
-        -- print progress
-        xlua.progress(startId, numThreads);
-
-        -- grab a batch
-        local batch, nextStartId =
-                        dataloader:getTestBatch(startId, self.params, dtype);
-
-        -- Call retrieve function for specific model, and store ranks
-        ranks[{{startId, nextStartId - 1}, {}}] = self:retrieveBatch(batch);
-        startId = nextStartId;
-    end
+    local retrievalOutput;
+    if dtype == 'val' then
```

Comment:
Should this be for both 'val' and 'train', just in case someone wants to compute retrieval metrics on training set?

Reply:
Okay that makes sense, I'll do `dtype ~= 'test'`

Reply:
Oh, this is on an outdated diff too -- all these changes reflect when I wrote the code according to my first approach. I changed it to separate methods and this line got lost. Although I will make a minor change in upcoming commit.